trade-finance-manager-ui/server/controllers/case/amendments/amendmentTask.controller.test.js

```diff
@@ -0,0 +1,202 @@
+          taskId: '1',
+        },
+      };
+
+      await postAmendmentTask(modifiedRequest, mockResponse);
+
+      expect(console.error).toHaveBeenLastCalledWith(
+        'Unable to update amendment task %s for group %s %o',
+        modifiedRequest.params.taskId,
+        modifiedRequest.params.groupId,
+        new Error('Invalid mandatory parameter'),
+      );
+
+      expect(mockResponse.redirect).toHaveBeenCalledWith('_partials/problem-with-service.njk');
+    });
+
+    it('when facility id is invalid', async () => {
+      const modifiedRequest = {
+        ...mockRequest,
+        params: {
+          _id: MOCKS.MOCK_DEAL._id,
+          amendmentId: MOCKS.MOCK_AMENDMENT.amendmentId,
+          groupId: '1',
+          taskId: '1',
+        },
+      };
+
+      await postAmendmentTask(modifiedRequest, mockResponse);
+
+      expect(console.error).toHaveBeenLastCalledWith(
+        'Unable to update amendment task %s for group %s %o',
+        modifiedRequest.params.taskId,
+        modifiedRequest.params.groupId,
+        new Error('Invalid mandatory parameter'),
+      );
+
+      expect(mockResponse.redirect).toHaveBeenCalledWith('_partials/problem-with-service.njk');
+    });
+
+    it('when amendment id is invalid', async () => {
+      const modifiedRequest = {
+        ...mockRequest,
+        params: {
+          _id: MOCKS.MOCK_DEAL._id,
+          facilityId: MOCKS.MOCK_AMENDMENT.facilityId,
+          groupId: '1',
+          taskId: '1',
+        },
+      };
+
+      await postAmendmentTask(modifiedRequest, mockResponse);
+
+      expect(console.error).toHaveBeenLastCalledWith(
+        'Unable to update amendment task %s for group %s %o',
+        modifiedRequest.params.taskId,
+        modifiedRequest.params.groupId,
+        new Error('Invalid mandatory parameter'),
+      );
+
+      expect(mockResponse.redirect).toHaveBeenCalledWith('_partials/problem-with-service.njk');
+    });
+
+    it('when task group id is invalid', async () => {
+      const modifiedRequest = {
+        ...mockRequest,
+        params: {
+          _id: MOCKS.MOCK_DEAL._id,
+          facilityId: MOCKS.MOCK_AMENDMENT.facilityId,
+          amendmentId: MOCKS.MOCK_AMENDMENT.amendmentId,
+          taskId: '1',
+        },
+      };
+
+      await postAmendmentTask(modifiedRequest, mockResponse);
+
+      expect(console.error).toHaveBeenLastCalledWith(
+        'Unable to update amendment task %s for group %s %o',
+        modifiedRequest.params.taskId,
+        modifiedRequest.params.groupId,
+        new Error('Invalid mandatory parameter'),
+      );
+
+      expect(mockResponse.redirect).toHaveBeenCalledWith('_partials/problem-with-service.njk');
+    });
+
+    it('when task id is invalid', async () => {
+      const modifiedRequest = {
+        ...mockRequest,
+        params: {
+          _id: MOCKS.MOCK_DEAL._id,
+          facilityId: MOCKS.MOCK_AMENDMENT.facilityId,
+          amendmentId: MOCKS.MOCK_AMENDMENT.amendmentId,
```

Comment:
Some repetition with these params, can we create a DRY `baseParams` or similar?

Reply:
Good spot, have rectified.